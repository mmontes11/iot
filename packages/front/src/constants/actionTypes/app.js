export const TOGGLE_HAMBURGER_MENU = "TOGGLE_HAMBURGER_MENU";
export const TOGGLE_MAP_DIALOG = "TOGGLE_MAP_DIALOG";
export const SET_SHOW_ERROR = "SET_SHOW_ERROR";
