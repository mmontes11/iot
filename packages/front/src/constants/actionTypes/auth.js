export const IS_AUTH = "IS_AUTH";
export const SET_USERNAME = "SET_USERNAME";
export const SET_PASSWORD = "SET_PASSWORD";
export const LOGIN_REQUEST = "LOGIN_REQUEST";
export const LOGIN_REQUEST_SUCCESS = "LOGIN_REQUEST_SUCCESS";
export const LOGIN_REQUEST_ERROR = "LOGIN_REQUEST_ERROR";
