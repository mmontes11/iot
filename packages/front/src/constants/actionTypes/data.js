export const DATA_UPDATED = "DATA_UPDATED";
export const DATA_REQUEST = "DATA_REQUEST";
export const DATA_REQUEST_SUCCESS = "DATA_REQUEST_SUCCESS";
export const DATA_REQUEST_ERROR = "DATA_REQUEST_ERROR";
export const ADD_DATA_ITEM = "ADD_DATA_ITEM";
export const RESET_DATA = "RESET_DATA";
