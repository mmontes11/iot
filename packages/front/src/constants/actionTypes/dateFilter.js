export const DATE_FILTER_SELECTOR_TOGGLE = "DATE_FILTER_SELECTOR_TOGGLE";
export const DATE_FILTER_TIME_PERIOD_SELECT = "DATE_FILTER_TIME_PERIOD_SELECT";
export const DATE_FILTER_TIME_PERIOD_UPDATED = "DATE_FILTER_TIME_PERIOD_UPDATED";
export const DATE_FILTER_TIME_PERIODS_REQUEST = "DATE_FILTER_TIME_PERIODS_REQUEST";
export const DATE_FILTER_TIME_PERIODS_REQUEST_SUCCESS = "DATE_FILTER_TIME_PERIODS_REQUEST_SUCCESS";
export const DATE_FILTER_TIME_PERIODS_REQUEST_ERROR = "DATE_FILTER_TIME_PERIODS_REQUEST_ERROR";
export const DATE_FILTER_TIME_PERIODS_UPDATED = "DATE_FILTER_TIME_PERIODS_UPDATED";
export const DATE_FILTER_START_DATE_UPDATED = "DATE_FILTER_START_DATE_UPDATED";
export const DATE_FILTER_END_DATE_UPDATED = "DATE_FILTER_END_DATE_UPDATED";
