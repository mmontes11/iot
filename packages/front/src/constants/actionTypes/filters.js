export const FILTER_TYPE_SELECT = "FILTER_TYPE_SELECT";
export const ADD_FILTER_TYPE = "ADD_FILTER_TYPE";
export const DELETE_FILTER_TYPE = "DELETE_FILTER_TYPE";
