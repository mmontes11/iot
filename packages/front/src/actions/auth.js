import {
  IS_AUTH,
  SET_USERNAME,
  SET_PASSWORD,
  LOGIN_REQUEST,
  LOGIN_REQUEST_SUCCESS,
  LOGIN_REQUEST_ERROR,
} from "constants/actionTypes/auth";
import { setShowError } from "actions/app";
import iotClient from "lib/iotClient";

export const isAuth = () => dispatch => {
  iotClient.authService
    .isAuth()
    .then(isAuthVal => dispatch({ type: IS_AUTH, isAuth: isAuthVal }))
    .catch(() => dispatch({ type: IS_AUTH, isAuth: false }));
};

export const setUsername = username => dispatch => {
  dispatch({ type: SET_USERNAME, username });
};

export const setPassword = password => dispatch => {
  dispatch({ type: SET_PASSWORD, password });
};

export const login = () => (dispatch, getState) => {
  dispatch({ type: LOGIN_REQUEST });
  const { username, password } = getState().auth;
  iotClient.authService.setCredentials(username, password);
  iotClient.authService
    .getToken()
    .then(response => {
      dispatch({
        type: LOGIN_REQUEST_SUCCESS,
        statusCode: response.statusCode,
        error: null,
      });
      isAuth()(dispatch);
    })
    .catch(error => {
      dispatch({
        type: LOGIN_REQUEST_ERROR,
        statusCode: error.statusCode,
        error,
      });
      setShowError(true)(dispatch);
    });
};

export const logout = () => dispatch => {
  iotClient.authService
    .logout()
    .then(() => {
      isAuth()(dispatch);
    })
    .catch(() => {
      isAuth()(dispatch);
    });
};
