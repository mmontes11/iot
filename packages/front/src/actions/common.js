import { RESET } from "constants/actionTypes/common";

export const reset = preserveError => dispatch => {
  dispatch({ type: RESET, preserveError });
};
