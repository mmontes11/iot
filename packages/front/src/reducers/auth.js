import { IS_AUTH, SET_PASSWORD, SET_USERNAME } from "constants/actionTypes/auth";
import { RESET } from "constants/actionTypes/common";

export const initialState = {
  isAuth: false,
  username: null,
  password: null,
};

export default (state = initialState, { type, isAuth, username, password }) => {
  switch (type) {
    case IS_AUTH:
      return { ...initialState, isAuth };
    case SET_USERNAME:
      return { ...state, username };
    case SET_PASSWORD:
      return { ...state, password };
    case RESET:
      return { ...initialState, isAuth: state.isAuth };
    default:
      return state;
  }
};
