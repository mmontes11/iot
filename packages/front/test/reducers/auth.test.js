import deepFreeze from "deep-freeze";
import authReducer from "reducers/auth";
import { IS_AUTH, SET_USERNAME, SET_PASSWORD } from "constants/actionTypes/auth";
import { RESET } from "constants/actionTypes/common";
import { initialState } from "../constants/index";

const authInitialState = initialState.auth;
deepFreeze(authInitialState);

describe("reducers/auth", () => {
  it("has a default state", () => {
    expect(authReducer(undefined, { type: "@init" })).toMatchSnapshot();
  });
  it("reduces nothing", () => {
    expect(authReducer(authInitialState, { type: "WHATEVER" })).toMatchSnapshot();
  });
  it("reduces IS_AUTH", () => {
    expect(authReducer(authInitialState, { type: IS_AUTH, isAuth: true })).toMatchSnapshot();
  });
  it("reduces SET_USERNAME", () => {
    expect(authReducer(authInitialState, { type: SET_USERNAME, username: "USERNAME" })).toMatchSnapshot();
  });
  it("reduces SET_PASSWORD", () => {
    expect(authReducer(authInitialState, { type: SET_PASSWORD, password: "PASSWORD" })).toMatchSnapshot();
  });
  it("reduces RESET", () => {
    expect(authReducer(authInitialState, { type: RESET })).toMatchSnapshot();
  });
});
