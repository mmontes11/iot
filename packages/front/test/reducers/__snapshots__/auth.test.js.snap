// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers/auth has a default state 1`] = `
Object {
  "isAuth": false,
  "password": null,
  "username": null,
}
`;

exports[`reducers/auth reduces IS_AUTH 1`] = `
Object {
  "isAuth": true,
  "password": null,
  "username": null,
}
`;

exports[`reducers/auth reduces RESET 1`] = `
Object {
  "isAuth": false,
  "password": null,
  "username": null,
}
`;

exports[`reducers/auth reduces SET_PASSWORD 1`] = `
Object {
  "isAuth": false,
  "password": "PASSWORD",
  "username": null,
}
`;

exports[`reducers/auth reduces SET_USERNAME 1`] = `
Object {
  "isAuth": false,
  "password": null,
  "username": "USERNAME",
}
`;

exports[`reducers/auth reduces nothing 1`] = `
Object {
  "isAuth": false,
  "password": null,
  "username": null,
}
`;
